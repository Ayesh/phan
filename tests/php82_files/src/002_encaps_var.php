<?php
namespace NS978;

class Example {
    /** @var string */
    public $foo;
}
$obj = new Example();
const foo = 'x';
$x = 'abc';
$deprecated = [
    "${x}",
    "${x[0]}",
    "${(foo)}",
];
'@phan-debug-var $deprecated';
