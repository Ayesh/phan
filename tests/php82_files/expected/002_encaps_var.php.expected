%s:6 PhanUnreferencedPublicProperty Possibly zero references to public property \NS978\Example->foo
%s:11 PhanDebugAnnotation @phan-debug-var requested for variable $deprecated - it has union type array{0:'abc',1:string,2:'abc'}(real=array{0:'abc',1:string,2:'abc'})
%s:12 PhanDeprecatedEncapsVar Saw deprecated encapsulated string variable syntax for $x: Using ${var} in strings is deprecated, use {$var} instead
%s:13 PhanDeprecatedEncapsVar Saw deprecated encapsulated string variable syntax for $x[0]: Using ${var} in strings is deprecated, use {$var} instead
%s:14 PhanDeprecatedEncapsVar Saw deprecated encapsulated string variable syntax for ${foo}: Using ${expr} in strings is deprecated, use {${expr}} instead
